scripts/cloud-check-control/cloudAtlasCheckControl.py

```diff
@@ -0,0 +1,663 @@
+        self.atlasCheckLogName = "atlasCheck.log"
+        self.atlasCheckLog = os.path.join(self.atlasLogDir, self.atlasCheckLogName)
+        self.atlasConfig = atlasConfig
+
+        self.client = None
+        self.instanceName = "AtlasChecks"
+
+    @property
+    def s3InBucket(self):
+        return self.s3InFolder.strip("/").split("/")[0]
+
+    @property
+    def atlasInDir(self):
+        return os.path.join(self.homeDir, self.s3InFolder)
+
+    def atlasCheck(self):
+        """Submit an spark job to perform atlas checks on an EC2 instance.
+
+        If the CloudAtlasChecksControl includes an instance ID then atlas checks will
+        be executed on that instance. If no instance ID is defined then it will
+        create a new instance.
+
+        Dependencies:
+          - self.instanceId - indicates a running instance or "" to create one
+          - self.S3Atlas - indicates the S3 bucket and path that contains atlas files
+        """
+        logger.info("Start Sharding Process...")
+        if self.instanceId == "":
+            self.create_instance()
+            self.get_instance_info()
+
+        if not self.is_process_running("SparkSubmit"):
+            cmd = "mkdir -p {} {}".format(self.atlasLogDir, self.atlasOutDir)
+            if self.ssh_cmd(cmd):
+                finish("Unable to create directory {}".format(cmd), -1)
+
+            # remove the success or failure files from any last run.
+            if self.ssh_cmd("rm -f {}/_*".format(self.atlasOutDir)):
+                finish("Unable to clean up old status files", -1)
+
+            # mount the s3 bucket
+            cmd = "mount |grep ~/{}".format(self.s3InBucket)
+            if not self.ssh_cmd(cmd, quiet=True):
+                cmd = "sudo umount ~/{}".format(self.s3InBucket)
+                if self.ssh_cmd(cmd, quiet=True):
+                    finish("Unable to unmount S3 bucket", -1)
+            else:
+                cmd = "mkdir -p ~/{}".format(self.s3InBucket)
+                if self.ssh_cmd(cmd):
+                    finish("Unable to create directory {}".format(cmd), -1)
+
+            cmd = "s3fs {0} ~/{0} -o ro,umask=222,gid=1000,uid=1000 -o use_cache=/tmp/atlas/ -o iam_role=auto".format(
+                self.s3InBucket
+            )
+            if self.ssh_cmd(cmd):
+                finish("Unable to mount S3 bucket", -1)
+
+            atlasConfig = self.atlasConfig
+            # if configuration.json is a file then copy it to the EC2 instance
+            if self.atlasConfig.find("http") < 0:
+                atlasConfig = "file://" + os.path.join(self.homeDir, self.atlasConfig)
+
+            # sync local input folder with S3 input bucket and directory
+            # cmd = "aws s3 sync s3://{} {}".format(self.s3InFolder, self.atlasInDir)
+            # if self.ssh_cmd(cmd):
+            #     finish("Unable to pull atlas input files from S3", -1)
+
+            # TODO Call command to submit job
```

Comment:
Are these comments and TODO still useful? Consider deleting if not.

Reply:
Ooops. removed.